faker/providers/misc/__init__.py

```diff
@@ -130,12 +130,16 @@ def locale(self):
     def language_code(self):
         return self.random_element(Provider.language_locale_codes.keys())
 
-    def uuid4(self):
+    def uuid4(self, cast_to=str):
         """
         Generates a random UUID4 string.
+        @param cast_to: String. What date type you would like the UUID to be cast to. Default is string.
```

Comment:
`cast_to` is supposed to be a callable, and the default is the built-in callable `str`

Reply:
Im not too familiar with docstring syntax and best practices. How should I document the parameter correctly?

Reply:
You can use the [Sphinx syntax](http://www.sphinx-doc.org/en/1.4.8/domains.html#info-field-lists), like this:

```
:param cast_to:  What type you would like the UUID to be cast to. Default is `str`
:type cast_to: callable, None
```